### ISSUE TITLE
fprint None returns {}

### ISSUE THREAD

**Issue by PhE** [QQuick/Transcrypt#494]
Got a strange behaviour when fprinting a None value with this code : 

```
def test_fprint_none():
    # Works
    a = 1
    print(f'a={a}')
    assert f'a={a}' == 'a=1'

    # Fails
    a = None
    print(f'a={a}')
    assert f'a={a}' == 'a=None'
```

Output from CPyhton : 
```
a=1 
a=None
```

Output from Transcrypt : 
```
a=1 
a={}
Uncaught ... assert ...
```

**Comment by neriusmika** [comment]
Jacques, there is a bug with None values in string formatting, and it's easy to fix (\_\_builtin\_\_.mod.js):
```javascript
            return self.replace (/\{(\w*)\}/g, function (match, key) {
                if (key == '') {
                    key = autoIndex++;
                }
                if (key == +key) {  // So key is numerical
                    return args [key] === undefined ? match : str (args [key]);  <== fix this line
                }
                else {              // Key is a string
                    for (var index = 0; index < args.length; index++) {
                        // Find first 'dict' that has that key and the right field
                        if (typeof args [index] == 'object' && args [index][key] != undefined) {
                            return str (args [index][key]); // Return that field field
                        }
                    }
                    return match;
                }
            });
```

This bug exists with -sf switch too, but I will look at it later.

**Comment by JdeH** [comment]
I'd say conversion to None would be best,
since that's what e.g. functions without return statement return in Python.
Open to better ideas, however...

**Comment by JdeH** [comment]
Fixed in 3.7.12
